app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java

```diff
@@ -154,12 +160,13 @@ private void storeStudentImage(Mat img){
 
         MatName matName = new MatName(sId, img);
         FileHelper fh = new FileHelper();
-        String wholeFolderPath = MultimediaHelper.getStudentImageDirectory() + "/" + deviceId + "/" + collectionEventId;
+        String wholeFolderPath = MultimediaHelper.getStudentImageDirectory() + "/" + deviceId + "/" + sId;
         new File(wholeFolderPath).mkdirs();
         fh.saveMatToImage(matName, wholeFolderPath + "/");
 
         Long Id =  Long.parseLong(String.valueOf((int) (Math.random() * 1000000)));
         StudentImage studentImage = new StudentImage(Id, null, wholeFolderPath, null, null);
```

Comment:
@jogrimst Sorry I'm wrong I thought I can save the DeviceID to long but it turns out the DeviceID is 20 digits long and the Long only takes 19. So we have to change it to String on the DB.